andela/travel_tool_deployment_scripts
# #166322780 Set Object Lifecycle Management Policy

#### What does this PR do?
This PR creates a bucket policy to change the storage class of objects in a bucket depending on their age.

#### Description of Task to be completed?
- Create a bucket policy rules in JSON
- Automate the setting of the rules using a bash script

#### How should this be manually tested?
- Clone this repo: `git clone https://github.com/andela/travel_tool_deployment_scripts.git`
- Change directory to the project: `cd travel_tool_deployment_scripts`
- Checkout to this branch: `git checkout  ft-add-bucket-policy-rules-166322780`
- Create a .env file from the `sample.env` file in the `bucketPolicy` directory. Replacing the placeholders with the appropriate values.
- Make the scripts executable: `chmod +x bucketPolicy/utils.sh  bucketPolicy/gcloud-setup.sh`
- Run the scripts: `sh bucketPolicy/gcloud-setup.sh`
- Check your GCP console to ensure that the policy has been applied to your bucket.

#### Any background context you want to provide?
The bucket policy will help with cost saving. Different storage classes have different costs, moving the less used objects to the cheaper options will save the organisation money on storage.

#### What are the relevant pivotal tracker stories?
[#166322780](https://www.pivotaltracker.com/story/show/166322780)

#### Screenshots (if appropriate)
N/A

#### Questions:
